src/payload/collections/Nominations/access/beforeVoting.ts

```diff
@@ -1,15 +1,23 @@
 import type { FieldAccess } from 'payload/types'
 
-import type { Election, Nomination } from '../../../payload-types'
+import type { Nomination } from '../../../payload-types'
 import { checkRole } from '../../Users/checkRole'
 
-export const beforeVoting: FieldAccess<Nomination> = ({ req: { user }, doc }) => {
-  if (user && checkRole(['admin'], user)) {
+export const beforeVoting: FieldAccess<Nomination> = async ({ req, doc }) => {
+  if (req.user && checkRole(['admin'], req.user)) {
     return true
   }
 
-  const election = doc.election as Election
-  const votingStart = Date.parse(election.votingStart)
+  // Don't think doc.election will ever not be a string, but just in case
+  const election = typeof doc.election === 'string' ? doc.election : doc.election.id
```

Comment:
I wrote a function called `getID` to help solve such problems

Reply:
I can see, let me update